packages/financial-templates-lib/src/price-feed/DefiPulseTotalPriceFeed.js

```diff
@@ -0,0 +1,146 @@
+const { PriceFeedInterface } = require("./PriceFeedInterface");
+const assert = require("assert");
+
+// An implementation of PriceFeedInterface that uses DefiPulse Data api to retrieve prices.
+class DefiPulseTotalPriceFeed extends PriceFeedInterface {
+  /**
+   * @notice Constructs the DefiPulseTVL_ALLPriceFeed.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} web3 Provider from truffle instance to connect to Ethereum network.
+   * @param {String} apiKey DeFiPulse Data API key. Note: these API keys are rate-limited.
+   * @param {Integer} lookback How far in the past the historical prices will be available using getHistoricalPrice.
+   * @param {Object} networker Used to send the API requests.
+   * @param {Function} getTime Returns the current time.
+   * @param {Integer} minTimeBetweenUpdates Min number of seconds between updates. If update() is called again before
+   *      this number of seconds has passed, it will be a no-op.
+   */
+  constructor(logger, web3, apiKey, lookback, networker, getTime, minTimeBetweenUpdates) {
+    super();
+    this.logger = logger;
+    this.web3 = web3;
+
+    this.apiKey = apiKey;
+    assert(apiKey, "requires apiKey for Defi Pulse");
+
+    this.lookback = lookback;
+    this.networker = networker;
+    this.getTime = getTime;
+    this.minTimeBetweenUpdates = minTimeBetweenUpdates;
+
+    this.toWei = this.web3.utils.toWei;
+
+    this.historicalPrices = [];
+  }
+
```

Comment:
@chrismaree can correct me if I'm wrong, but I think you should make the `decimals` value (i.e. how many decimals you scale your output to) configurable with a constructor parameter.

You also need to add a `getPriceFeedDecimals()` method that returns the number of decimals that the price feed is outputting (which is a part of the PriceFeedInterface on master: https://github.com/UMAprotocol/protocol/blob/ec5908e93c72e547b94b2ccf675b3c5d94ad7b93/packages/financial-templates-lib/src/price-feed/PriceFeedInterface.js#L37).